probe-rs/src/core/mod.rs

```diff
@@ -366,6 +386,11 @@ pub trait CoreInterface: MemoryInterface {
     /// This must be queried while halted as this is a runtime
     /// decision for some core types
     fn instruction_set(&mut self) -> Result<InstructionSet, error::Error>;
+
+    /// Determine if an FPU is present.
+    /// This must be queried while halted as this is a runtime
+    /// decision for some core types.
+    fn fpu_support(&mut self) -> Result<bool, error::Error>;
```

Comment:
I'm trying to reason about if adding a top level core function like this makes sense versus a property on the register file.  I don't have a strong opinion here, but I know that adding every feature as a `CoreInterface` function like `supports_x` may be hard to scale.

The only alternative I can think of is to return a different `RegisterFile` based the core state.  For example, return the current one with no FP regs on cores that don't have the FP bit set, or return a new ARM_REGISTERS_WITH_FP on ones that do.

I *think* I slightly prefer what you have, but others may have different opinions.

Reply:
@Yatekii , @Tiwalun  - do either of you have an opinion on this?

Reply:
To me it is uncertain how many such extension we'll need in the future. Atm it feels like there is not many. But this could be a case of lacking foresight.

Do you think there will be many more features like this we need to add?

Reply:
Another solution (for this specific case) that would be viable is to include this data in the instruction set info. That is also [read from the device](https://github.com/probe-rs/probe-rs/blob/aaaa9dbf6c436de97a24553eda2c01063e8301f0/probe-rs/src/core/mod.rs#L385-L388).
So, like [here](https://github.com/probe-rs/probe-rs/blob/9c14a09f74ce9899ee3480b525e49e62c3f0f8ae/probe-rs-target/src/chip_family.rs#L78), you could add an FPU boolean field for every instruction set where it would be relevant. (All of them right now)

Reply:
The only other case like this I can think of is SIMD registers.  One more supports function doesn't feel too bad, which is part of why I was leaning towards the solution as proposed.  I can think of lots of arch specific flags, but that needs modeled a different way no matter what.

I don't like this on the instruction set as it's not a property of the ISA, it's a property of the core and its register file.

Reply:
Ok, yeah SIMD was the only thing I was thinking of too. I think in the end the `RegisterFile` needs an adaptation anyways, since not all cores will have FP, so a function to grab them would return an `Option` I guess. Having multiple types for different architectures might be unnecessary trouble since you need to match everytime to even access general purpose registers every core has.

Reply:
Agreed.  Given that I think the path to close this is to move the `fp_support` calls on non-implemented platforms back to an error based on the feedback and then we're ready for a final review / merge.

Reply:
Ok, great. I'll revert that then :)

Reply:
Ok, done

Reply:
@Yatekii - I think this is ready for a final review / merge